tools/Az.Tools.Predictor/Az.Tools.Predictor/ParameterSet.cs

```diff
@@ -68,11 +74,22 @@ public ParameterSet(CommandAst commandAst)
 
             Parameters = parameters;
 
-            void AddParameter(CommandParameterAst parameterName, Ast parameterValue)
+            void AddParameter(CommandParameterAst parameter, Ast parameterValue)
             {
-                if (parameterName != null)
+                if (parameter != null)
                 {
-                    parameters.Add(new Parameter(parameterName.ParameterName, (parameterValue == null) ? null : CommandLineUtilities.UnescapePredictionText(parameterValue.ToString())));
+                    var value = parameterValue?.ToString();
+                    if (value == null)
+                    {
+                        value = parameter.Argument?.ToString();
+                    }
+
+                    if (value != null)
```

Comment:
I just want to make sure you want the first if to be checking `parameterValue?.ToString();` and the second if to be checking if `parameter.Argument?.ToString()`?

Reply:
The first is to check `parameterValue?.ToString()`. If that's null, then I'll set `value` to `parameter.Argument?.ToString()`. So the second `if` checks whether the `value` is null. `value` is either from `parameterValue` or `parameter.Argument`.

There are two formats of parameter name and value pair here.
1. "-name value". The value is in `parameterValue`.
2. "-name:value". The value is in `parameter.Argument`.

Reply:
OK, just wanted to make sure since it looked like something that could be mixed up.